### PR TITLE
Milestone 5

### PR DESCRIPTION
**Pull request by verissimohenry**
the last task is to tie the calculations module with the components and to add the interaction to the buttons.

### CODE REVIEW

**Review by mahmoud717 — CHANGES_REQUESTED** [review]
### Changes required ♻️ 

**Hello @verissimohenry**

Great job with the project so far it looks amazing, but there are still some changes to be made:
 
---
### Requested 🔴 

- [x] Please make sure to add division by zero error. Right now when I divide by zero nothing happens. Please make sure to fix that,

---

please make these changes and request another code review.

if you have any questions don't hesitate to contact me on slack **MahmoudMohammad**

**Review by taiwo2 — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED  ♻️ 
@verissimohenry well-done on this milestone, what commendable efforts on trying to implement the suggested changes, however, changes are required
Please make sure to display  `division by zero error`. when divide by zero.

### once specific changes are made kindly request for review

File: src/logic/operate.js [hunk 1]
```diff
@@ -1,3 +1,6 @@
+/* eslint-disable consistent-return */
+/* eslint-disable no-unreachable */
+/* eslint-disable no-undef */
```
Comment (bellom) [inline src/logic/operate.js]: With the ESLint inline comment, you are indirectly changing the linter's rules by disabling these rules, let's fix the codebase instead and remove the inline comments.

File: src/logic/operate.js [hunk 2]
```diff
@@ -28,8 +31,12 @@ export default function operate(previous, next, operation) {
     case '=':
       value = prev;
       break;
+    case '÷':
+      if (currentNum === 0) this.error = "Can't divide by zero";
+      else calculation = previousNum / currentNum;
```
Comment (bellom) [inline src/logic/operate.js]: const used in here seem undefined 🤔 

Hint: [Microverse operate.js template](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/code-samples/operate.js)

image


With the above example, kindly fix the `eslint` and add division by zero error message.

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @verissimohenry :clap: 

**Great work making `almost` all the necessary corrections from the previous review.**

# Status: Changes Required! ♻️ 

**1. Let's make the division by zero error message logic is working properly, for now, it is not.**

**2. Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Hi, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️

- [x] Your APP crashed when I do a division by 0. Please fix it:


- [x] Your APP return ` Not valid operation` when I use +/- and *.
- [x] I got an error related to propType when I do an operation.



_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

- [x] Make sure that your APP is styled based on Milestone 3.
- [x] Files not related to the project should not be committed and added to .gitignore (e.g. .vscode) 
([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/gitignore.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by christianotieno — APPROVED** [review]
Hi @verissimohenry, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

**Pull request merged**